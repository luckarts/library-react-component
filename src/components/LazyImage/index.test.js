import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { LazyImage } from './index';

let wrapper, props;
props = {
	src: 'https://bachelart.fr/img/study_color2.jpg',
	alt: 'study_color2',
	placeholderSrc: 'https://bachelart.fr/thumb/img/study_color2.jpg',
};

describe('Galerie Component with children props', () => {
	beforeEach(() => {
		wrapper = shallow(<LazyImage {...props} />);
	});
	it('renders an img without a src attribute', () => {
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
describe('when it mounts', () => {
	const mockRef = { mock: true };
	const windowIntersectionObserver = window.IntersectionObserver;
	const observe = jest.fn();

	beforeAll(() => {
		window.IntersectionObserver = jest.fn(function() {
			this.observe = observe;
		});

		wrapper = shallow(<LazyImage {...props} />, {
			disableLifecycleMethods: true,
		});
		wrapper.getElement().ref(mockRef);
		wrapper.instance().componentDidMount();
	});
});
describe('with additional properties', () => {
	beforeAll(() => {
		wrapper = shallow(<LazyImage {...props} />, {
			disableLifecycleMethods: true,
		});
	});

	it('passes them to the img element', () => {
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
