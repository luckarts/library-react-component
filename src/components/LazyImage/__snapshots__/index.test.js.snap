// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Galerie Component with children props renders an img without a src attribute 1`] = `
<img
  alt="study_color2"
  src="https://bachelart.fr/img/study_color2.jpg"
/>
`;

exports[`with additional properties passes them to the img element 1`] = `
<img
  alt="study_color2"
  src="https://bachelart.fr/img/study_color2.jpg"
/>
`;
